src/signer_node/mod.rs

```diff
@@ -131,120 +131,145 @@ impl<T: TapyrusApi, C: ConnectionManager> SignerNode<T, C> {
+                Err(std::sync::mpsc::TryRecvError::Empty) => {
+                    // Stop signal is empty. Continue to run. Do nothing.
+                    None
                 }
+                Err(_) => Some(()),
+            },
+            None => {
+                // Stop signal receiver is not set. Do nothing.
+                None
             }
-            // Checking network connection error
-            match connection_manager_error_handler {
-                Some(ref receiver) => match receiver.try_recv() {
-                    Ok(e) => {
-                        self.round_timer.stop();
-                        log::error!("Connection Manager Error {:?}", e);
-                        panic!(e.to_string());
-                    }
-                    Err(TryRecvError::Empty) => {
-                        // No errors.
-                    }
-                    Err(e) => log::debug!("{:?}", e),
-                },
-                None => {
-                    log::warn!("Failed to get error_handler of connection_manager!");
+        }
+    }
+
+    fn handle_message(&mut self, receiver: &Receiver<Message>) {
+        // Receiving message.
+        match receiver.try_recv() {
+            Ok(Message {
+                message_type,
+                sender_id,
+                ..
+            }) => {
+                log::debug!(
+                    "Got {} message from {:?}. MessageType: {:?}",
+                    message_type,
+                    sender_id,
+                    message_type
+                );
+
+                let next = self.process_round_message(&sender_id, message_type);
+                self.current_state = next;
+
+                if let NodeState::RoundComplete { .. } = &self.current_state {
+                    self.start_next_round()
                 }
+
+                log::debug!("Current state updated as {:?}", self.current_state);
+            }
+            Err(TryRecvError::Empty) => {
+                // No new messages. Do nothing.
             }
+            Err(e) => {
+                log::warn!("Can't receive message: {:?}", e);
+            }
+        }
+    }
 
-            // Wait for next loop 300 ms.
-            std::thread::sleep(Duration::from_millis(300));
+    fn handle_timer(&mut self) {
+        // Checking whether the time limit of a round exceeds.
+        match self.round_timer.receiver.try_recv() {
+            Ok(_) => {
+                // Round duration is timeout. Starting next round.
+                self.start_next_round();
+                log::debug!("Current state updated as {:?}", self.current_state);
+            }
+            Err(TryRecvError::Empty) => {
+                // Still waiting round duration interval. Do nothing.
+            }
+            Err(e) => log::warn!("Round timer generates an error: {:?}", e),
         }
     }
 
+    fn handle_connection(&mut self) -> Option<ConnectionManagerError<C::ERROR>> {
```

Comment:
How about `handle_connection_error`? 
Because this method will try to handle errors.

Reply:
f60d2baa787d39bf33571a03f890a3ab7b498401